plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -65,6 +65,9 @@ public class IcebergTableHandle
     private final Set<IcebergColumnHandle> projectedColumns;
     private final Optional<String> nameMappingJson;
 
+    // Coordinator-only - table partitioning applied to the table splits
+    private final Optional<IcebergTablePartitioning> requiredTablePartitioning;
```

Comment:
The name is misleading. Coordinator should be still free to ignore `requiredTablePartitioning` if `io.trino.sql.planner.plan.TableScanNode#isUseConnectorNodePartitioning` is false.

Reply:
This was a WIP.  In the final version this is just named tablePartitioning.